rust/main/agents/relayer/src/msg/op_queue.rs

```diff
@@ -72,27 +72,66 @@ impl OpQueue {
         // The other consideration is whether to put the channel receiver in the OpQueue or in a dedicated task
         // that also holds an Arc to the Mutex. For simplicity, we'll put it in the OpQueue for now.
         let mut message_retry_requests = vec![];
-        while let Ok(message_id) = self.retry_rx.lock().await.try_recv() {
-            message_retry_requests.push(message_id);
+
+        // we only need to lock self.retry_receiver once
+        {
+            let mut retry_receiver = self.retry_receiver.lock().await;
+            while let Ok(retry_request) = retry_receiver.try_recv() {
+                let uuid = retry_request.uuid.clone();
+                message_retry_requests.push((
+                    retry_request,
+                    MessageRetryResponse {
+                        uuid,
+                        evaluated: 0,
+                        matched: 0,
+                    },
+                ));
+            }
         }
         if message_retry_requests.is_empty() {
             return;
         }
         let mut queue = self.queue.lock().await;
+        let queue_length = queue.len();
+
         let mut reprioritized_queue: BinaryHeap<_> = queue
             .drain()
             .map(|Reverse(mut op)| {
-                if message_retry_requests.iter().any(|r| r.op_matches(&op)) {
-                    info!(
-                        operation = %op,
-                        queue_label = %self.queue_metrics_label,
-                        "Retrying OpQueue operation"
-                    );
-                    op.reset_attempts()
+                let mut matched = false;
+                message_retry_requests
+                    .iter_mut()
+                    .for_each(|(retry_req, retry_response)| {
+                        if !retry_req.pattern.op_matches(&op) {
+                            return;
+                        }
+                        debug!(
+                            uuid = retry_req.uuid,
+                            ?op,
+                            pattern = ?retry_req.pattern, "Matched retry request");
```

Comment:
Added more informative debug messages so we know which requests and patterns are being matched.
I hope this isn't the reason why the endpoint is slow now